packages/core/orchestration/src/workflow/scheduler.ts

```diff
@@ -0,0 +1,45 @@
+import { MedusaError } from "@medusajs/utils"
+import { IDistributedSchedulerStorage, SchedulerOptions } from "../transaction"
+import { WorkflowDefinition } from "./workflow-manager"
+
+export class WorkflowScheduler {
+  private static keyValueStore: IDistributedSchedulerStorage
+  public static setStorage(storage: IDistributedSchedulerStorage) {
+    this.keyValueStore = storage
+  }
+
+  public async scheduleWorkflow(workflow: WorkflowDefinition) {
+    const schedule = workflow.options?.schedule
+    if (!schedule) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_ARGUMENT,
+        "Workflow schedule is not defined while registering a scheduled workflow"
+      )
+    }
+
+    const normalizedSchedule: SchedulerOptions =
+      typeof schedule === "string"
+        ? {
+            cron: schedule,
+            concurrency: "forbid",
+          }
+        : {
+            cron: schedule.cron,
+            concurrency: schedule.concurrency || "forbid",
+            numberOfExecutions: schedule.numberOfExecutions,
+          }
+
+    await WorkflowScheduler.keyValueStore.scheduleJob(
+      workflow.id,
+      normalizedSchedule
+    )
+  }
+
+  public async clearWorkflow(workflow: WorkflowDefinition) {
```

Comment:
sugg: this can just accept the id directly right?

Reply:
What I was thinking is that maybe we want to have some options set on the workflow in terms of how clearing works, so I opted for passing the entire object. I am happy to change it to id only as well, I think for now either is fine